genie-agent/src/main/java/com/netflix/genie/agent/cli/CommandFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.cli;
+
+import com.beust.jcommander.Parameters;
+import org.springframework.context.ApplicationContext;
+
+import java.util.List;
+
+/**
+ * Factory to create a (lazy) bean for the requested AgentCommand.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+class CommandFactory {
+    private final List<AgentCommandArguments> agentCommandArgumentsBeans;
+    private final ApplicationContext applicationContext;
+
+    CommandFactory(
+        final List<AgentCommandArguments> agentCommandArguments,
+        final ApplicationContext applicationContext
+    ) {
+        this.agentCommandArgumentsBeans = agentCommandArguments;
+        this.applicationContext = applicationContext;
+    }
+
+    AgentCommand get(final String requestedCommandName) {
```

Comment:
Is there no better way to do this than with nested loops? not that there are many commands but we can't hash them or anything?